test/src/module/storage/remoteTest.c

```diff
@@ -452,17 +466,35 @@ testRun(void)
         ((StorageRemote *)storageDriver(storageRepoWrite))->compressLevel = 3;
 
         // -------------------------------------------------------------------------------------------------------------------------
-        TEST_TITLE("write file, free before close, make sure the .tmp file remains");
+        TEST_TITLE("write file, free before close, make sure the .tmp file remains (interleaved with normal write)");
 
+        StorageWrite *write3 = NULL;
         TEST_ASSIGN(write, storageNewWriteP(storageRepoWrite, STRDEF("test2.txt")), "new write file");
+        TEST_ASSIGN(write3, storageNewWriteP(storageRepoWrite, STRDEF("test3.txt")), "new write file");
 
         TEST_RESULT_VOID(ioWriteOpen(storageWriteIo(write)), "open file");
+        TEST_RESULT_VOID(ioWriteOpen(storageWriteIo(write3)), "open file 3");
+
         TEST_RESULT_VOID(ioWrite(storageWriteIo(write), contentBuf), "write bytes");
+        TEST_RESULT_VOID(ioWrite(storageWriteIo(write3), contentBuf), "write bytes to file 3");
 
         TEST_RESULT_VOID(storageWriteFree(write), "free file");
+        TEST_RESULT_VOID(ioWriteClose(storageWriteIo(write3)), "close file 3");
 
         TEST_RESULT_UINT(
             storageInfoP(storageTest, STRDEF("repo128/test2.txt.pgbackrest.tmp")).size, 16384, "file exists and is partial");
+        TEST_RESULT_BOOL(bufEq(storageGetP(storageNewReadP(storageRepo, STRDEF("test3.txt"))), contentBuf), true, "check file 3");
+
+        // -------------------------------------------------------------------------------------------------------------------------
+        TEST_TITLE("write free and close before write");
+
+        TEST_ASSIGN(write, storageNewWriteP(storageRepoWrite, STRDEF("test2.txt")), "new write file");
+        TEST_RESULT_VOID(ioWriteOpen(storageWriteIo(write)), "open file");
+        TEST_RESULT_VOID(storageWriteFree(write), "free file");
+
+        TEST_ASSIGN(write, storageNewWriteP(storageRepoWrite, STRDEF("test2.txt")), "new write file");
+        TEST_RESULT_VOID(ioWriteOpen(storageWriteIo(write)), "open file");
+        TEST_RESULT_VOID(ioWriteClose(storageWriteIo(write)), "close file");
 
```

Comment:
Well, we can't interleave reads and writes to the same file if that's what you mean. In general that is not a defined action since most of our writes are atomic.

Do you just mean read/write two separate files in an interleaved fashion? We could add that but not sure if it really provides any coverage.

Reply:
I can add the tests if you think it is important. I'm not sure it is -- doing multiple reads and multiple writes shows that the read/writes don't interfere with other current processes. There are lots of possible combinations, i.e. read/list path, write/remove path, etc. There are a lot of combos so I don't see what makes read/write special.

Reply:
I'm fine with your instincts here.